machine/machine.go

```diff
@@ -156,6 +185,34 @@ func getCoreIdFromCpuBus(cpuBusPath string, threadId int) (int, error) {
 	return int(coreId), nil
 }
 
+/* Look for sysfs cpu path containing given CPU property, e.g. core_id or physical_package_id */
+/* and returns number of unique values of given property, exemplary usage: getting number of CPU physical cores */
+func getUniqueCPUPropertyCount(cpuBusPath string, propertyName string) int {
+	uniquePropertyCount := 0
+	pathPattern := cpuBusPath + "cpu*[0-9]"
+	sysCPUPaths, err := filepath.Glob(pathPattern)
+	if err != nil {
+		klog.Errorf("Cannot find files matching pattern (path-pattern: %s),  number of unique %s set to %d", pathPattern, propertyName, uniquePropertyCount)
+		return uniquePropertyCount
+	}
+	uniques := make(map[string]bool)
+	for _, sysCPUPath := range sysCPUPaths {
+		propertyPath := filepath.Join(sysCPUPath, "topology", propertyName)
+		propertyVal, err := ioutil.ReadFile(propertyPath)
+		propertyStr := string(propertyVal)
+		if err != nil {
+			klog.Errorf("Cannot open %s, number of unique %s  set to %d", propertyPath, propertyName, uniquePropertyCount)
+			return uniquePropertyCount
+		}
+
+		if !uniques[propertyStr] {
```

Comment:
Consider removing `if` clause - a map with assure that keys are unique anyway.

Reply:
Exactly! Nice catch!